lib/explorer/backend/lazy_frame.ex

```diff
@@ -0,0 +1,55 @@
+defmodule Explorer.Backend.LazyFrame do
+  @moduledoc """
+  Represents a dataframe in an opaque lazy format.
+  """
+
+  alias Explorer.Backend
+  alias Explorer.Backend.LazySeries
+
+  defstruct dtypes: %{}, names: [], original: nil
+
+  @type t :: %__MODULE__{
+          dtypes: Backend.DataFrame.dtypes(),
+          names: Backend.DataFrame.column_name()
+        }
+  @behaviour Backend.DataFrame
+
+  @doc false
+  def new(df) do
+    %__MODULE__{names: df.names, dtypes: df.dtypes, original: df}
+  end
+
+  @impl true
+  def lazy, do: __MODULE__
+
+  @impl true
+  def to_lazy(ldf), do: ldf
+
+  @impl true
+  def inspect(ldf, opts) do
+    Backend.DataFrame.inspect(ldf.data.original, "OpaqueLazyFrame", nil, opts)
```

Comment:
TODO: We likely want to inspect this as a query expression, instead of the original data frame? I am unsure... I would personally not keep the original dataframe at all.